tests/fsharp/Compiler/CompilerAssert.fs

```diff
@@ -102,6 +106,37 @@ let main argv = 0"""
                 try Directory.Delete(projectDirectory) with | _ -> ()
 #endif
 
+#if FX_NO_APP_DOMAINS
```

Comment:
This is good. Means we can build this test project as any TFM without worrying about runtime configs and copying FSharp.Core/etc.